streams/src/test/java/org/apache/kafka/streams/integration/PurgeRepartitionTopicIntegrationTest.java

```diff
@@ -87,7 +88,7 @@ public static void closeCluster() {
     }
 
 
-    private final Time time = CLUSTER.time;
+    private final Time time = new MockTime(1);
```

Comment:
The time needs to progress when the state updater is shutdown, otherwise the shutdown is blocked. `CLUSTER` provides a mock time that does not progress.

Reply:
Are we not worried that broker time and our time might diverge? Can we make the broker mock time advance? Seems to be an undesired pattern to have two mock-times?

Reply:
I tried to expose the `autoTickMs` parameter in embedded kafka, but I ran into a infinite loop.

Regarding your question:

> Are we not worried that broker time and our time might diverge?

In almost all our integration tests broker time diverges from streams time since the broker time does not advance but we usually use system time in `KafkaStreams`. This is not something this PR introduces.